packages/mui-material-next/src/MenuItem/MenuItem.js

```diff
@@ -190,30 +202,37 @@ const MenuItem = React.forwardRef(function MenuItem(inProps, ref) {
     dense: childContext.dense,
     divider,
     disableGutters,
+    disabled,
+    focusVisible,
+    highlighted,
   };
 
   const classes = useUtilityClasses(props);
 
-  const handleRef = useForkRef(menuItemRef, ref);
-
   let tabIndex;
   if (!props.disabled) {
     tabIndex = tabIndexProp !== undefined ? tabIndexProp : -1;
   }
 
+  const Root = /* slots.root ?? */ MenuItemRoot;
+  const rootProps = useSlotProps({
+    elementType: Root,
+    getSlotProps: getRootProps,
+    // externalSlotProps: slotProps.root,
+    externalForwardedProps: other,
+    additionalProps: {
+      role,
+      tabIndex,
+      component,
+      focusVisibleClassName: clsx(classes.focusVisible, focusVisibleClassName),
+      classes,
+    },
+    className: clsx(classes.root, className),
+    ownerState,
+  });
   return (
     <ListContext.Provider value={childContext}>
```

Comment:
Do we need `ListContext` still?

Reply:
Until we migrate to Material You I am not sure, it contains some styling-related props, like `dense`, `disableGutters` etc. I would keep them for now